txscript/data/script_tests.json

```diff
@@ -1365,87 +1459,193 @@
+
+["Equivalency of different numeric encodings"],
+["0x02 0x8000", "128 NUMEQUAL", "NONE", "OK", "0x8000 equals 128"],
+["0x01 0x00", "0 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x00 numequals 0 but rejected by minimal data encoding requirements"],
+["0x01 0x80", "0 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x80 (negative zero) numequals 0 but rejected by minimal data encoding requirements"],
+["0x02 0x0080", "0 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x0080 numequals 0 but rejected by minimal data encoding requirements"],
+["0x02 0x0500", "5 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0x0500 numequals 5 but rejected by minimal data encoding requirements"],
+["0x03 0xff7f80", "0x02 0xffff NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xff7f80 numequals 0xfffff but rejected by minimal data encoding requirements"],
+["0x03 0xff7f00", "0x02 0xff7f NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xff7f00 numequals 0xff7f but rejected by minimal data encoding requirements"],
+["0x04 0xffff7f80", "0x03 0xffffff NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xffff7f80 numequals 0xffffff but rejected by minimal data encoding requirements"],
+["0x04 0xffff7f00", "0x03 0xffff7f NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "0xffff7f00 numequals 0xffff7f but rejected by minimal data encoding requirements"],
+
+["NUMEQUALVERIFY test coverage"],
+["NOP NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items and NOP must not be treated as one"],
+["NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items, stack has no items"],
+["0 NUMEQUALVERIFY", "1", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUALVERIFY requires two stack items, stack has one item"],
+["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK", "NUMEQUALVERIFY must compare the first numeric stack item to the second numeric stack item, check if they are equal and verify the result"],
+["-1 -1", "NUMEQUALVERIFY 1", "NONE", "OK"],
+["1 2147483648", "NUMEQUALVERIFY 1", "NONE", "ERR_OUT_OF_RANGE", "NUMEQUALVERIFY must fail with stack item >4 bytes"],
+
+["NUMNOTEQUAL test coverage"],
+["NOP", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items and NOP must not be treated as one"],
+["", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items, stack has no items"],
+["0", "NUMNOTEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMNOTEQUAL requires two stack items, stack has one item"],
+["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK", "NUMNOTEQUAL must compare the first numeric stack item to the second numeric stack item and check if they are not equal" ],
 ["111 10 1 ADD", "NUMNOTEQUAL", "NONE", "OK"],
-["11 10", "LESSTHAN NOT", "NONE", "OK"],
+["1 -1", "NUMNOTEQUAL", "NONE", "OK"],
+["1 2147483648", "NUMNOTEQUAL", "NONE", "ERR_OUT_OF_RANGE", "NUMNOTEQUAL must fail with stack item >4 bytes"],
```

Comment:
Same here regarding needing another test for the other operand being >4 bytes and comment.